lib/msal-core/docs/scopes.md

```diff
@@ -0,0 +1,88 @@
+
+### OpenID Connect Scopes for Authentication
+
+`OpenID Connect (OIDC) scopes` are a specific set of scopes that can be added to requests when `authenticating` a user. In most cases, `OIDC scopes` are added to configure the claims included in an ID Token ([OIDC Reference](https://openid.net/specs/openid-connect-core-1_0.html#ScopeClaims) / [Microsoft Docs](https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-permissions-and-consent#openid-connect-scopes)). Some services, such as the Secure Token Service that `msal@1.x` acquires tokens from, also use OIDC scopes in their internal logic. For this reason, it is important to understand and pay attention to the special behavior `msal@1.x` has around OIDC scopes (described in the [next section](#default-scopes-on-authorization-requests)).
+
+The OIDC scopes that `msal@1.x` pays particular attention to are outlined in the table below.
+
+| OIDC Scope | Required by OIDC Specification | Function | OIDC Reference | Microsoft Docs |
+| ---------- | ------------------------------ | -------- | -------------- | -------------- |
+| `openid`| Mandatory   |  Main `OIDC scope` that indicates a request for `authentication` [per the OIDC specification](https://openid.net/specs/openid-connect-core-1_0.html#AuthRequest). In AAD requests, this is the scope that prompts the "Sign in" permission that a user can consent to. | [Authentication Request](https://openid.net/specs/openid-connect-core-1_0.html#AuthRequest)| [Permissions and consent in the Microsoft identity platform endpoint](https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-protocols-oidc#send-the-sign-in-request)|
+|`profile`| Optional | Used for ID Token `claims` configuration. Adds the end-user's default profile information as a claim to the ID token returned | [Requesting Claims using Scopes Values](https://openid.net/specs/openid-connect-core-1_0.html#ScopeClaims) | [OpenID Permissions](https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-permissions-and-consent)|
+
+
+## Scopes Behavior
+
+### Default Scopes on Authorization Requests
+
+Understanding how `OIDC scopes` configure the claims included in an authentication response's ID Token is important when using `msal@1.x` to acquire said ID Tokens. However, there is an important note to be made on how the `openid` and `profile` scopes are added by `msal@1.x` to all server requests by default that does not directly relate to the OpenID Connect specification.
+
+Like previously mentioned, the Secure Token Service that `msal@1.x` requests access and ID tokens from also makes use of the `openid` and `profile` scopes. Specifically, the STS expects these two scopes in order to configure and provide the `client_info` parameter in authorization and authentication responses. The `msal@1.x` library depends on the contents of `client_info` in order to successfuly cache tokens and, therefore, provide silent token acquisition as a feature.
```

Comment:
`client_info` is internal detail. Would we want this in documentation? I prefer not to. cc @tnorling

Reply:
I tried not to go too into the weeds, but these docs are for developers at the end of the day. I think anyone trying to change this in the future should know why it was done.

Reply:
Others may disagree but, I personally feel that docs should be as much for the customers as they are for us/new members of the team so I don't have an issue with sharing implementation details as long as customers are still easily able to find what they're looking for without wading through stuff they don't care about. Maybe to accomplish this we include a quick TLDR summary at the top of each doc that just contains the important points and then we go into detail below?

Reply:
I think the main purpose of this specific set of docs is to bring clarity to ambiguous features/behaviors. Even while trying to be concise, I don't feel comfortable obfuscating the very implementation details that made this such a complex issue in the first place.

Reply:
I totally agree, but I think you can have a summary that distills it down to a few important points without explaining the why if someone only cares about "how do I do x?" For example for the scopes doc the 2 most important things to take away are: 
- Msal.js will always add `openid` and `profile to every request
- ClientId sent as the only scope will be replaced by `openid` and `profile` and return an id_token

For the response_type doc:
- Login requests will always return only an id_token
- AcquireToken requests will always return an id_token if `openid` or `profile` are included in the request
- AcquireToken requests will always return an access_token if a resource scope is requested

Sure there are more nuances and more details and those should stay. But I think these quick points will answer 75% of people's questions without needing to read the whole doc.

Reply:
Added TLDRs at the top of each doc as a "Quick Reference" section with a small indication that those are the key takeaways and people should read on for more info/clarification. Thanks for the suggestions!